### ISSUE TITLE
What's Happening filter

### ISSUE THREAD

**Issue by sfederici** [ScratchAddons/ScratchAddons#4124]
### Is your request related to a problem? Please describe.

The possibility to infinitely scroll the what's happening box is fantastic to me. But there are too many items to go through. I'm mainly interested to favorited projects

### Describe the solution you'd like

Allow to filter the items in the What's Happening box, e.g. loved, favorited, created, followed, etc

### Describe alternatives you've considered

No alternative considered

### Additional context

None

**Comment by lisa-wolfgang** [comment]
Related: https://github.com/ScratchAddons/ScratchAddons/issues/3115, https://github.com/ScratchAddons/ScratchAddons/issues/1758

**Comment by wxterfaII** [comment]
yes, please do this.

**Comment by Norbiros** [comment]
It can't work like in "messages filter" addon, because when you create project and love project it shows message with the same class: "love-project", so you can't detect them in these way.

**Comment by sfederici** [comment]
Is it at least possible to discriminate loved from favorited from follows from became a curator of?

**Comment by Norbiros** [comment]
Yes. Using simple code I found 7 categories. Yes you can do that.

**Comment by sfederici** [comment]
Excellent. That would be great.

**Comment by Norbiros** [comment]
Update, this was Scratch's bug. In API everything works great (you have more categories), but in html some classes are duplicated. So it is possible to discriminate between created project and loved project!

**Comment by sfederici** [comment]
> So it is possible to discriminate between created project and loved project!

Excellent!

**Comment by Norbiros** [comment]
I think that we can create something like this:
![Screenshot from 2022-05-20 11-16-53](https://user-images.githubusercontent.com/89837724/169496300-8b7e629a-a48d-4364-a68c-fc01b9722654.png)
All elements has tooltips (when you hover on it it shows name).
Also when some category is hide,  icon colour is gray...

What do you think about this idea?

**Comment by sfederici** [comment]
> What do you think about this idea?

If I correctly understood, the elements I don't need will be grayed out? To me is fine, even if, completely removing them, would be even better

**Comment by Norbiros** [comment]
No. Icon will be grayed out, but messages will be hidden. 
In messages-filter we have a lot of space, so you can add check-box, but in "what's happening" box icon is checkbox (when it is clicked it has color, but when it is unclicked it is gray).

**Comment by WorldLanguages** [comment]
@Norbiros There should be a way to toggle the categories inside the settings page. What you're showing could be done as a bonus, so users can temporarily show other categories until they reload.

**Comment by Norbiros** [comment]
> @Norbiros There should be a way to toggle the categories inside the settings page. What you're showing could be done as a bonus, so users can temporarily show other categories until they reload.

IMAO This addon should work like `messages-filter` - you have gui in messages page, and it saves what you selected to local storage. If we want to implement way to change settings in GUI and in settings page, we need to have #2483

**Comment by WorldLanguages** [comment]
> IMAO This addon should work like `messages-filter`

I'm aware that addon works differently, but doesn't it have like 15 types of messages? How many could "What's Happening" show? I bet it's less than 10?

**Comment by sfederici** [comment]
> > IMAO This addon should work like `messages-filter`
> 
> I'm aware that addon works differently, but doesn't it have like 15 types of messages? How many could "What's Happening" show? I bet it's less than 10?

I'm not sure I understood your comment fully. If you are mentioning the 10 lines in the "What's Happening"  box, I use the "infinite scrolling" addon, that allows me to see in that box as many lines I want (I use to scroll it until I get 3-9 hours before the current time).

**Comment by mxmou** [comment]
> > > IMAO This addon should work like `messages-filter`
> > 
> > 
> > I'm aware that addon works differently, but doesn't it have like 15 types of messages? How many could "What's Happening" show? I bet it's less than 10?
> 
> I'm not sure I understood your comment fully. If you are mentioning the 10 lines in the "What's Happening" box, I use the "infinite scrolling" addon, that allows me to see in that box as many lines I want (I use to scroll it until I get 3-9 hours before the current time).

World_Languages is talking about the number of different types of items that one might want to show or hide. There are only eight, but I don't think that makes the settings page a good place to choose which items should be shown.

**Comment by sfederici** [comment]
> World_Languages is talking about the number of different types of items that one might want to show or hide. 

I see

> There are only eight, but I don't think that makes the settings page a good place to choose which items should be shown.

Why not?

**Comment by WorldLanguages** [comment]
+1 from feedback

**Comment by WorldLanguages** [comment]
+1 from feedback

> So baisically I'd like to filter it so I don't see those messages and only see when they post something.
